crates/solvers/src/run.rs

```diff
@@ -28,47 +29,63 @@ async fn run_with(args: cli::Args, bind: Option<oneshot::Sender<SocketAddr>>) {
     observe::tracing::initialize_reentrant(&args.log);
     tracing::info!("running solver engine with {args:#?}");
 
-    let solver = match args.command {
+    let solver = match &args.command {
         cli::Command::Baseline { config } => {
-            let config = config::baseline::file::load(&config).await;
+            let config = config::baseline::file::load(config).await;
             Solver::Baseline(solver::Baseline::new(config))
         }
         cli::Command::Naive { config } => {
-            let config = config::naive::file::load(&config).await;
+            let config = config::naive::file::load(config).await;
             Solver::Naive(solver::Naive::new(config))
         }
         cli::Command::Legacy { config } => {
-            let config = config::legacy::load(&config).await;
+            let config = config::legacy::load(config).await;
             Solver::Legacy(solver::Legacy::new(config))
         }
         cli::Command::ZeroEx { config } => {
-            let config = config::dex::zeroex::file::load(&config).await;
+            let config = config::dex::zeroex::file::load(config).await;
             Solver::Dex(solver::Dex::new(
                 dex::Dex::ZeroEx(
                     dex::zeroex::ZeroEx::new(config.zeroex).expect("invalid 0x configuration"),
                 ),
                 config.base,
+                RateLimiter::new(
+                    args.single_order_solver_rate_limiter.unwrap_or_default(),
+                    args.command.to_lowercase(),
```

Comment:
nit: I don't think we need a solver specific name here. Solvers already run in separate containers, so just having a common name (e.g. "dex_api") should be enough to differentiate.

Reply:
Replaced with the common name